src/Bicep.Decompiler/TemplateConverter.cs

```diff
@@ -480,7 +499,19 @@ private SyntaxBase ParseFunctionExpression(FunctionExpression expression)
                         }
 
                         var stringVal = jTokenExpression.Value.Value<string>()!;
-                        var resolved = nameResolver.TryLookupName(NameType.Variable, stringVal) ?? throw new ArgumentException($"Unable to find variable {stringVal}");
+                        var resolved = nameResolver.TryLookupName(NameType.Variable, stringVal);
```

Comment:
nit: This looks similar to code in ln 477. Can we refactor it to a separate method and re use?